homeassistant/components/sensor/geo_rss_events.py

```diff
@@ -0,0 +1,239 @@
+                                         unit_of_measurement)
+            devices.append(device)
+    add_devices(devices, True)
+
+
+class GeoRssServiceSensor(Entity):
+    """Representation of a Sensor."""
+
+    def __init__(self, category, data, service_name, unit_of_measurement):
+        """Initialize the sensor."""
+        self._category = category
+        self._data = data
+        self._service_name = service_name
+        self._state = STATE_UNKNOWN
+        self._state_attributes = None
+        self._unit_of_measurement = unit_of_measurement
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return '{} {}'.format(self._service_name,
+                              'Any' if self._category is None
+                              else self._category)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the default icon to use in the frontend."""
+        return DEFAULT_ICON
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes."""
+        return self._state_attributes
+
+    def update(self):
+        """Update this sensor from the GeoRSS service."""
+        _LOGGER.debug("About to update sensor %s", self.entity_id)
+        self._data.update()
+        if self._category is None:
+            # Add all events regardless of category.
+            my_events = self._data.events
+        else:
+            # Only keep events that belong to sensor's category.
+            my_events = [event for event in self._data.events if
+                         event[ATTR_CATEGORY] == self._category]
+        _LOGGER.debug("Adding events to sensor %s: %s", self.entity_id,
+                      my_events)
+        self._state = len(my_events)
+        # And now compute the attributes from the filtered events.
+        matrix = {}
+        for event in my_events:
+            matrix[event[ATTR_TITLE]] = '{:.0f}km'.format(
+                event[ATTR_DISTANCE])
+        self._state_attributes = matrix
+
+
+class GeoRssServiceData(object):
+    """Provides access to GeoRSS feed and stores the latest data."""
+
+    def __init__(self, home_latitude, home_longitude, url, radius_in_km):
+        """Initialize the update service."""
+        self._home_coordinates = [home_latitude, home_longitude]
+        self._url = url
+        self._radius_in_km = radius_in_km
+        self.events = None
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self):
+        """Retrieve data from GeoRSS feed and store events."""
+        import feedparser
+        feed_data = feedparser.parse(self._url)
+        if not feed_data:
+            _LOGGER.error("Error fetching feed data from %s", self._url)
```

Comment:
This case could lead to an error if `self.events` is still `None`, since you don't check that before taking the length of `my_events` in `GeoRssServiceSensor.update`. You don't seem to test this case either.

Reply:
Yep, you got me there - that was the single line that was not covered through tests. I fixed the issue, and added a new test case

Reply:
100% test coverage reached.